app/services/container_topology_service.rb

```diff
@@ -89,22 +89,26 @@ def entity_display_type(entity)
     end
   end
 
-  def build_entity_data(entity)
+  def entity_id(entity)
     type = entity_type(entity)
-    status = entity_status(entity, type)
-
     if entity.kind_of?(ManageIQ::Providers::ContainerManager)
       id = entity.id.to_s
     else
       id = case type
-           when 'Vm', 'Host'
-             entity.uid_ems
-           else
-             entity.ems_ref
+             when 'Vm', 'Host'
+               entity.uid_ems
+             else
+               entity.ems_ref
```

Comment:
> @abonas Can you instead specify some role in the models so you don't have to hardcode class names?

I really don't see what roles and why should be added to the models themselves. this is a very specific logic for topology. why touch the models for that?